### ISSUE TITLE
Client-side speed up techniques

### ISSUE THREAD

**Issue by pietercolpaert** [openplannerteam/planner.js#57]
This is a brainstorm issue about how we could speed up client-side query execution of Planner.js:

## Prefetch data on page load and store in memory

When the page loads, immediately start downloading the data for the now time frame.

When the data is downloaded, store the connections in a data structure that allows the CSA algorithm to be performed without having to set up an HTTP request.

Cache invalidation then works by refreshing the page.

## Web Workers

https://developer.mozilla.org/en-US/docs/Web/API/Web_Workers_API/Using_web_workers

Web Workers allow to do tasks in parallel and use the multiple cores. E.g., downloading the data per transport mode and executing the algorithm itself could be done in different threads.

## Other?

 * There are other ideas to speed up the client-side querying by adapting the server-interface. This is out of scope of this issue, but of course, the less data one needs to download, the more efficient the querying goes. Therefore we will look at Contraction Hierarchies techniques for PT networks as well.

 * Another server-side idea would be to switch to another serialization as JSON-LD parsing is a bottle neck. For now this is out of scope of the project: we want the project to be as flexible first. Being able to support any RDF serialization makes sure we can answer any question.
